text/0000-fabric-x-next-generation.md

```diff
@@ -0,0 +1,365 @@
+* **Configuration:** Configuration will be more granular, specific to each microservice. The genesis configuration will use the same config block format and `configtxgen` (or equivalent), with additional fields for the new ordering services. MSP, ACLs, and Policies remain conceptually the same.
+* **Debugging:** Troubleshooting will involve analyzing logs and interactions across multiple services.
+* **Performance:** Expect significantly higher throughput (demonstrated >150,000 TPS) and lower latency.
+* **Feature Set:** Be aware that Fabric-X will not initially support all HLF v2.x/v3.x features (e.g., Private Data Collections, Channels). The focus is on core transactional performance. A roadmap for reintroducing essential Fabric features will be developed post-initial release.
+* **Security:** While individual components are more isolated, the overall security of the distributed system, including inter-service communication, must be carefully managed. The compromise of a Query Service node does not inherently allow it to substitute data in the sharded distributed database if the database itself maintains strong consistency and write integrity is handled by the VCs; the Query Service primarily performs reads. Writes to the database are governed by the VCs after successful validation.
+* **Programming Model Shift:** Developers will transition from chaincode to 'Flows' using FSC or an endorser SDK. This requires learning but offers more power and flexibility.
+
+**Migration and Coexistence Strategy**
+
+A significant point of discussion is how Fabric-X relates to existing Hyperledger Fabric deployments and the path forward for users. Fabric-X is designed as a new, high-performance implementation within the Fabric ecosystem, not as a direct, drop-in replacement for all existing Fabric networks.
+
+* **No Direct Automated Migration (Initial Version):** Due to the fundamental architectural differences (microservices, distributed database per org, evolved programming model, absence of initial channel support), there is no direct, automated migration path from an existing Fabric network to Fabric-X in its initial version. Applications and network configurations would require re-architecture.
+* **Addressing Future Migration Needs:** Should the community express a strong preference and a clear need emerge for migrating specific existing Fabric use cases to Fabric-X (particularly those hitting performance ceilings), a migration path could be developed in future iterations.
```

Comment:
This basically says there is not clear migration path, highlighting the initial intent not to support migration.

Reply:
There will be no automatic migration to Fabric-X. However, assisted migration will be possible.

Fabric is a general-purpose platform. If its current performance is sufficient for existing use cases, users do not need to transition to Fabric-X.